### PR TITLE
Staker should be allowed to set comission before accepting delegation.

### LINKED ISSUE #372: setcommission should be done before accepting delegation.
setcommission should be done before accepting delegation.
staker can accept delegation and after people delegate, they can set commission too high.

### PR DESCRIPTION
**Pull request by adi44**
fixes #372

### CODE REVIEW

File: contracts/Core/StakeManager.sol [hunk 1]
```diff
@@ -224,7 +225,7 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
     function setCommission(uint8 commission) external {
         uint32 stakerId = stakerIds[msg.sender];
         require(stakerId != 0, "staker id = 0");
-        require(stakers[stakerId].acceptDelegation, "Delegetion not accpected");
+        require(!stakers[stakerId].acceptDelegation, "Delegation accepted");
```
Comment (abhishekvispute) [inline contracts/Core/StakeManager.sol:228]: Hey @adi44 
Is this require needed anymore ?
Without this well seq is assured
As you cant accept delegation unless you have setCommission
So if you are calling setCommission, its given you have not accepted delegation.

Comment (adi44) [reply contracts/Core/StakeManager.sol:228]: Yes true, we can remove that. it seems to be redundant.

File: test/StakeManager.js [hunk 2]
```diff
@@ -584,36 +584,21 @@ describe('StakeManager', function () {
...
 
-    it('Staker should not be able to set commission if delegation not accepted', async function () {
-      const tx = stakeManager.connect(signers[1]).setCommission(5);
-      await assertRevert(tx, 'Delegetion not accpected');
+    it('Staker should not be able to accept delegation if comission is not set', async function () {
+      const tx = stakeManager.connect(signers[1]).setDelegationAcceptance('true');
+      await assertRevert(tx, 'comission not set');
     });
-
     it('Delegator should not be able to delegate if delegation not accepted', async function () {
```
Comment (abhishekvispute) [inline test/StakeManager.js:591]: Hey 
did we miss adding this test case ?

File: contracts/Core/StakeManager.sol (hunk 1 above)
Comment (abhishekvispute) [reply contracts/Core/StakeManager.sol]: cool

File: test/StakeManager.js (hunk 2 above)
Comment (adi44) [reply test/StakeManager.js:591]: no all the testcases are there. but I had changed few things as per the new require condition

Comment (abhishekvispute) [reply test/StakeManager.js:591]: cool, got it

**Review by abhishekvispute — APPROVED** [review]
LGTM

**Review by SkandaBhat — APPROVED** [review]

**Pull request merged**